packages/@uppy/webdav/src/Webdav.tsx

```diff
@@ -0,0 +1,113 @@
+import { h } from 'preact'
+import { useState } from 'preact/hooks'
+
+import { UIPlugin } from '@uppy/core'
+import { Provider, tokenStorage } from '@uppy/companion-client'
+import { ProviderViews } from '@uppy/provider-views'
+
+import packageJson from '../package.json'
+import locale from './locale.ts'
+
+class WebdavSimpleAuthProvider extends Provider {
+  async login({ authFormData, uppyVersions, signal }) {
+    return this.loginSimpleAuth({ uppyVersions, authFormData, signal })
+  }
+
+  async logout() {
+    this.removeAuthToken()
+    return { ok: true, revoked: true }
+  }
+}
+
+const AuthForm = ({ loading, i18n, onAuth }) => {
+  const [webdavUrl, setWebdavUrl] = useState('')
+
+  const onSubmit = (event) => {
+    event.preventDefault()
+    onAuth({ webdavUrl: webdavUrl.trim() })
+  }
```

Comment:
usecallback to prevent unnecessary re-renders?
```suggestion
  const onSubmit = useCallback((e) => {
    e.preventDefault()
    onAuth({ webdavUrl: webdavUrl.trim() })
  }, [onAuth, webdavUrl])
```

Reply:
haha, @Murderlon told me to remove the useCallback we had in place before...

Reply:
`useCallback`/`useMemo` are overused and should only be leveraged if you _know_ something is slow, not just for the sake of it.

Reply:
when coding it's not always obvious that something is going to become slow once it's in production with real world data. IMO it's safer to just add a useCallback than to find out by surprise later that your app is re-rendering the whole React tree on every key stroke in an `<input>` for example.

Reply:
But this isn't every keystroke, this is a form submit. General advice (from the React team itself) is to not just add it for the sake of it, in fact it's often slower to do the memorization check than what it saves.

Anyway this is such a minor thing in a big PR like this, we better focus on something else.